### ISSUE TITLE
Peers with no TUN?

### ISSUE THREAD

**Issue by lonnietc** [yggdrasil-network/yggdrasil-go#1039]
Hello,

I read in the FAQ that you can run a peer without a TUN device is basically using the existing network connection from what I understand, right?

Also, if a person does this with one peer then they will not be able to communicate with other yggdrasil peers on the network which I would surmise that this is because those other peers are running with TUN devices.

My question is wondering if multiple nodes were set up with no TUN devices, then could those non-tun devices communicate with  each other by themselves in a type of non-tun mesh network?

Any thoughts would be greatly appreciated.
Thanks

**Comment by perguth** [comment]
No, not having an interface just means that the node will only route traffic for others but not send/receive packages itself.

**Comment by neilalexander** [comment]
The TUN device is needed to send IP traffic over Yggdrasil. If you just want to forward traffic on behalf of other nodes (i.e. as a pure router) then you can disable TUN. You won't be able to send or receive any Yggdrasil IPv6 traffic if you do that.

Whether or not TUN is enabled or disabled does not affect peerings in any way. You can peer Yggdrasil nodes regardless of whether they have TUN enabled or not.

**Comment by lonnietc** [comment]
@neilalexander Thanks for the comments on yggdrasil and I will test it more.   

I think that I actually still want to work with pinecone if I can get it to work and send some test messages. Maybe you can look at that one if you get a bit of time since what I really need is something like yggdrasil or pinecone that is an overlay network that does not need tun devices and uses existing network adapters for a small p2p project that I am working on now.

Thanks agian

**Comment by neilalexander** [comment]
Right now the only non-TUN option that either Yggdrasil or Pinecone can realistically support right now is to be embedded into a Go application (where both can be used as a normal `net.PacketConn`, like UDP sockets are) or bound to Java/Kotlin or Swift/Obj-C using `gomobile` bindings.

**Comment by lonnietc** [comment]
Hi Neal,

Actually, that is exactly what I am wanting to accomplish to embed pinecone or yggdrasil inside a Golang application that I want to build since I will need to add more components.  On the pinecone side I was thinking to take the cmd/pinecone/pinecone.go application or the cmd/yggdrasil/main.go and see if I could expand it to send messages to one node, multiple nodes, or all nodes with the untimate goal of having a private mesh network that for the peers to connect.  This particular project is a type of private p2p ecommerce application which will also allow me to get better with Go as I am coming from a C/C++ background but like many features that I have seen in Go, especially the cross-platform abilities.
